File: api/Enkl.Api/Controllers/ProjectStrategyController.cs
using Enkl.Api.Services;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace Enkl.Api.Controllers;

/// <summary>
/// Read-only Strategy surface for regular project members (ProjectMemberAuthorizationHandler already
/// live-verifies membership — or Org Admin status — against a real ProjectMembers row per request,
/// same discipline as PrinciplesController/RisksController/etc). No CRUD lives here at all; every
/// write happens through StrategyController (OrgAdmin) or the Portfolio Planner's fulfilment-upsert
/// route instead.
/// </summary>
[ApiController]
[Authorize(Policy = "ProjectMember")]
[Route("api/projects/{projectId:guid}/strategy")]
public class ProjectStrategyController : ControllerBase
{
    private readonly StrategyPillarService _pillars;
    private readonly StrategyMetricService _metrics;
    private readonly StrategyFulfilmentService _fulfilment;

    public ProjectStrategyController(StrategyPillarService pillars, StrategyMetricService metrics, StrategyFulfilmentService fulfilment)
    {
        _pillars = pillars;
        _metrics = metrics;
        _fulfilment = fulfilment;
    }

    [HttpGet("tree")]
    public async Task<IActionResult> GetTree(Guid projectId)
    {
        var tree = await _pillars.GetActivePillarTreeForProjectAsync(projectId);
        return tree is null ? NotFound() : Ok(tree);
    }

    [HttpGet("metrics/{metricId:guid}/entries")]
    public async Task<IActionResult> GetMetricHistory(Guid projectId, Guid metricId)
    {
        var history = await _metrics.GetHistoryForProjectAsync(projectId, metricId);
        return history is null ? NotFound() : Ok(history);
    }

    [HttpGet("fulfilment")]
    public async Task<IActionResult> GetFulfilment(Guid projectId)
    {
        var matrix = await _fulfilment.BuildSingleProjectMatrixAsync(projectId);
        return matrix is null ? NotFound() : Ok(matrix);
    }
}

File: mariadb-api/src/Controllers/ProjectStrategyController.php
<?php

declare(strict_types=1);

namespace Enkl\Api\Controllers;

use Enkl\Api\Db\Database;
use Enkl\Api\Services\StrategyFulfilmentService;
use Enkl\Api\Services\StrategyMetricService;
use Enkl\Api\Services\StrategyPillarService;
use Psr\Http\Message\ResponseInterface as Response;
use Psr\Http\Message\ServerRequestInterface as Request;

/**
 * Ported from php-api's Controllers/ProjectStrategyController.php (itself ported from
 * Controllers/ProjectStrategyController.cs). Read-only Strategy surface for regular project members
 * (ProjectMemberMiddleware, gated in routes.php). No CRUD lives here at all.
 */
final class ProjectStrategyController extends BaseController
{
    private function pillars(): StrategyPillarService
    {
        return new StrategyPillarService(Database::connection());
    }

    private function metrics(): StrategyMetricService
    {
        return new StrategyMetricService(Database::connection());
    }

    private function fulfilment(): StrategyFulfilmentService
    {
        return new StrategyFulfilmentService(Database::connection());
    }

    public function getTree(Request $request, Response $response, array $args): Response
    {
        $tree = $this->pillars()->getActivePillarTreeForProject($args['projectId']);
        return $tree !== null ? $this->json($response, $tree) : $this->notFound($response);
    }

    public function getMetricHistory(Request $request, Response $response, array $args): Response
    {
        $history = $this->metrics()->getHistoryForProject($args['projectId'], $args['metricId']);
        return $history !== null ? $this->json($response, $history) : $this->notFound($response);
    }

    public function getFulfilment(Request $request, Response $response, array $args): Response
    {
        $matrix = $this->fulfilment()->buildSingleProjectMatrix($args['projectId']);
        return $matrix !== null ? $this->json($response, $matrix) : $this->notFound($response);
    }
}
